it/v1/endpoints/RetrieveEmploymentsExpensesControllerISpec.scala

```diff
@@ -16,267 +16,69 @@
-         |    "source": "LATEST",
-         |    "totalExpenses": 123.12,
-         |    "expenses": {
-         |        "businessTravelCosts": 123.12,
-         |        "jobExpenses": 123.12,
-         |        "flatRateJobExpenses": 123.12,
-         |        "professionalSubscriptions": 123.12,
-         |        "hotelAndMealExpenses": 123.12,
-         |        "otherAndCapitalAllowances": 123.12,
-         |        "vehicleExpenses": 123.12,
-         |        "mileageAllowanceRelief": 123.12
-         |    }
-         |}
-         |""".stripMargin)
-
-    val hmrcHeldDesResponseBody = Json.parse(s"""
-         |{
-         |    "submittedOn": "2020-12-12T12:12:12Z",
-         |    "source": "HMRC HELD",
-         |    "totalExpenses": 123.12,
-         |    "expenses": {
-         |        "businessTravelCosts": 123.12,
-         |        "jobExpenses": 123.12,
-         |        "flatRateJobExpenses": 123.12,
-         |        "professionalSubscriptions": 123.12,
-         |        "hotelAndMealExpenses": 123.12,
-         |        "otherAndCapitalAllowances": 123.12,
-         |        "vehicleExpenses": 123.12,
-         |        "mileageAllowanceRelief": 123.12
-         |    }
-         |}
-         |""".stripMargin)
-
-    val userDesResponseBody = Json.parse(s"""
-         |{
-         |    "submittedOn": "2020-12-12T12:12:12Z",
-         |    "source": "CUSTOMER",
-         |    "totalExpenses": 123.12,
-         |    "expenses": {
-         |        "businessTravelCosts": 123.12,
-         |        "jobExpenses": 123.12,
-         |        "flatRateJobExpenses": 123.12,
-         |        "professionalSubscriptions": 123.12,
-         |        "hotelAndMealExpenses": 123.12,
-         |        "otherAndCapitalAllowances": 123.12,
-         |        "vehicleExpenses": 123.12,
-         |        "mileageAllowanceRelief": 123.12
-         |    }
-         |}
-         |""".stripMargin)
-
-    def latestUri: String   = s"/employments/$nino/$taxYear?source=$latestSource"
-    def hmrcHeldUri: String = s"/employments/$nino/$taxYear?source=$hmrcHeldSource"
-    def userUri: String     = s"/employments/$nino/$taxYear?source=$userSource"
-    def ifsUri: String      = s"/income-tax/expenses/employments/$nino/$taxYear"
-
-    def setupStubs(): StubMapping
-
-    def request(uri: String): WSRequest = {
-      setupStubs()
-      buildRequest(uri)
-        .withHttpHeaders(
-          (ACCEPT, "application/vnd.hmrc.1.0+json"),
-          (AUTHORIZATION, "Bearer 123") // some bearer token
-        )
-    }
-
-    def errorBody(code: String): String =
-      s"""
-         |{
-         |  "failures": [
-         |    {
-         |      "code": "$code",
-         |      "reason": "if message"
-         |    }
-         |  ]
-         |}
-    """.stripMargin
-
-  }
-
   "Calling the retrieve endpoint" should {
-
     "return a 200 status code" when {
+      "valid latest request is made" in new NonTysTest {
```

Comment:
Do we need individual tests for the different query param values?

Reply:
The query params aren't related to TYS so I didn't see any value in testing all the combinations.

Reply:
Agree wrt TYS. I meant do we need to have tests for all the query param values in the first place (non-TYS)? Could we get away with just having e.g. `hmrcHeld` and not test the others?

Reply:
Ah sorry misunderstood what you meant! I don't think there's any harm in having them, helps to double check the source string mappings work correctly etc, e.g. that `hmrcHeld` maps to `HMRC-HELD` for example.